README.md

```diff
@@ -2,32 +2,259 @@
+
 
 client = Client(client_id='***', secret='***')
-connect = client.connect(account_type='bofa', username='***', password='***', email='john@whatever.com')
+account_type = 'bofa'
 
-if connect.ok:
-    json_response = json.loads(connect.content)
+try:
+    response = client.connect(account_type, {
+     'username': '***',
+     'password': '***'
+    })
+except plaid_errors.PlaidError, e:
+     pass
+else:
+    connect_data = json.loads(response.content)
+```
+
+### MFA add Connect User
+
+```python
+
+from plaid import Client
+from plaid import errors as plaid_errors
+from plaid.utils import json
 
-    print json_response['mfa'][0] # Should be something like "What's your mother's maiden name?"
 
-    step = client.step(account_type='bofa', mfa='Smith')
-    if step.ok:
-        transactions = json.loads(step.content)
-        # ...
+client = Client(client_id='***', secret='***')
+account_type = 'bofa'
+
+try:
+    response = client.connect(account_type, {
+     'username': '***',
+     'password': '***'
+    })
+except plaid_errors.PlaidError, e:
+     pass
+else:
+    if response.status_code == 200:
+        # User connected
+        data = json.loads(response.content)
+    elif response.stat_code == 201:
+        # MFA required
+        try:
+            mfa_response = answer_mfa(json.loads(response.content))
+        except plaid_errors.PlaidError, e:
+            pass
+        else:
+            # check for 200 vs 201 responses
+            # 201 indicates that additional MFA steps required
+
+
+def answer_mfa(data):
+    if data['type'] == 'questions':
+        # Ask your user for the answer to the question[s].
+        # Although questions is a list, there is only ever a
+        # single element in this list, at present
+        return answer_question([q['question'] for q in data['mfa']])
+    elif data['type'] == 'list':
+        return answer_list(data['mfa'])
+    elif data['type'] == 'selection':
+        return answer_selections(data['mfa'])
+    else:
+        raise Exception('Unknown mfa type from Plaid')
+
+
+def answer_question(questions):
+    # We have magically inferred the answer
+    # so we respond immediately
+    # In the real world, we would present questions[0]
+    # to our user and submit their response
+    answer = 'dogs'
+    return client.connect_step(account_type, answer)
+
+
+def answer_list(devices):
+    # You should specify the device to which the passcode is sent.
+    # The available devices are present in the devices list
+    return client.connect_step('bofa', None, options={
+        'send_method': {'type': 'phone'}
+    })
+
+
+def answer_selections(selections):
+    # We have magically inferred the answers
+    # so we respond immediately
+    # In the real world, we would present the selection
+    # questions and choices to our user and submit their responses
+    # in a JSON-encoded array with answers provided
+    # in the same order as the given questions
+    answer = '["Yes", "No"]'
```

Comment:
Sure.  This was purely an example of the desired format but I agree that that is better.